diff -rupN original/bfd/config.bfd new/bfd/config.bfd
--- original/bfd/config.bfd	2013-11-04 16:33:37.000000000 +0100
+++ new/bfd/config.bfd	2013-12-25 19:51:28.656619916 +0100
@@ -538,6 +538,9 @@ case "${targ}" in
   i[3-7]86-*-aros*)
     targ_defvec=bfd_elf32_i386_vec
     ;;
+  i[3-7]86-*-utopia*)
+    targ_defvec=bfd_elf32_i386_vec
+    ;;
   i[3-7]86-*-chorus*)
     targ_defvec=bfd_elf32_i386_vec
     ;;
diff -rupN original/config.sub new/config.sub
--- original/config.sub	2013-11-26 12:37:33.000000000 +0100
+++ new/config.sub	2013-12-25 19:50:10.037045835 +0100
@@ -1356,7 +1356,7 @@ case $os in
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* | -plan9* \
 	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
-	      | -aos* | -aros* \
+	      | -aos* | -aros* | -utopia* \
 	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
 	      | -clix* | -riscos* | -uniplus* | -iris* | -rtu* | -xenix* \
 	      | -hiux* | -386bsd* | -knetbsd* | -mirbsd* | -netbsd* \
diff -rupN original/gas/configure.tgt new/gas/configure.tgt
--- original/gas/configure.tgt	2013-11-04 16:33:37.000000000 +0100
+++ new/gas/configure.tgt	2013-12-25 19:57:02.168146462 +0100
@@ -206,6 +206,7 @@ case ${generic_target} in
   i386-*-beos*)				fmt=elf ;;
   i386-*-coff)				fmt=coff ;;
   i386-*-elf)				fmt=elf ;;
+  i386-*-utopia*)			fmt=elf ;;
   i386-*-kaos*)				fmt=elf ;;
   i386-*-bsd*)				fmt=aout em=386bsd ;;
   i386-*-nacl*)				fmt=elf em=nacl
diff -rupN original/ld/configure.tgt new/ld/configure.tgt
--- original/ld/configure.tgt	2013-11-26 12:37:33.000000000 +0100
+++ new/ld/configure.tgt	2013-12-25 19:53:26.489314895 +0100
@@ -218,6 +218,7 @@ i[3-7]86-*-lynxos*)	targ_emul=i386lynx ;
 i[3-7]86-*-coff)	targ_emul=i386coff ;;
 i[3-7]86-*-rtems*)	targ_emul=elf_i386 ;;
 i[3-7]86-*-aros*)	targ_emul=elf_i386 ;;
+i[3-7]86-*-utopia*)	targ_emul=utopia_i386 ;;
 i[3-7]86-*-rdos*)	targ_emul=elf_i386 ;;
 x86_64-*-rdos*)		targ_emul=elf64rdos ;;
 i[3-7]86-*-bsd)		targ_emul=i386bsd ;;
diff -rupN original/ld/emulparams/utopia_i386.sh new/ld/emulparams/utopia_i386.sh
--- original/ld/emulparams/utopia_i386.sh	1970-01-01 01:00:00.000000000 +0100
+++ new/ld/emulparams/utopia_i386.sh	2013-12-25 19:54:38.118926844 +0100
@@ -0,0 +1,15 @@
+SCRIPT_NAME=elf
+OUTPUT_FORMAT="elf32-i386"
+NO_RELA_RELOCS=yes
+TEXT_START_ADDR=0x40000000
+MAXPAGESIZE="CONSTANT (MAXPAGESIZE)"
+COMMONPAGESIZE="CONSTANT (COMMONPAGESIZE)"
+ARCH=i386
+MACHINE=
+NOP=0x90909090
+TEMPLATE_NAME=elf32
+GENERATE_SHLIB_SCRIPT=yes
+GENERATE_PIE_SCRIPT=yes
+NO_SMALL_DATA=yes
+SEPARATE_GOTPLT=12
+IREL_IN_PLT=
diff -rupN original/ld/Makefile.in new/ld/Makefile.in
--- original/ld/Makefile.in	2013-11-26 12:37:33.000000000 +0100
+++ new/ld/Makefile.in	2013-12-25 19:55:37.035274333 +0100
@@ -2752,6 +2752,9 @@ eelf32xtensa.c: $(srcdir)/emulparams/elf
 eelf_i386.c: $(srcdir)/emulparams/elf_i386.sh \
   $(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
 	${GENSCRIPTS} elf_i386 "$(tdir_elf_i386)"
+eutopia_i386.c: $(srcdir)/emulparams/utopia_i386.sh \
+  $(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
+	${GENSCRIPTS} utopia_i386 "$(tdir_utopia_i386)"
 eelf_i386_be.c: $(srcdir)/emulparams/elf_i386_be.sh \
   $(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
 	${GENSCRIPTS} elf_i386_be "$(tdir_elf_i386_be)"
